docs/dev/agent_build.md

```diff
@@ -53,7 +53,21 @@ We use `pkg-config` to make compilers and linkers aware of Python. If you need
 to adjust the build for your specific configuration, add or edit the files within
 the `pkg-config` folder.
 
-The Agent is comprised of several binaries, each with its own invoke task to build it:
+By default, the Agent combines multiple functionalities into a single binary to reduce
+the space used on disk. The `DD_BUNDLED_AGENT` environment variable is used to select
+which functionality to enable. For instance, if set to `process-agent`, it will act as the process Agent.
+If the environment variable is not defined, the process name is used as a fallback.
+As the last resort meaning, the executable will behave as the 'main' Agent.
+
+Different combinations can be obtained through the usage of build tags. As an example,
+building the Agent with the `bundle_process_agent` and `bundle_security_agent` will produce
+a binary that has the process Agent and security Agent capabilities.
+
+The `--bundle` argument can be used to override the default set of functionalities bundled
```

Comment:
Since we are changing how to build the agent and how to run it, I would imagine that a lot of us are very internalized, so the change in workflow might be unexpected. I think adding a couple of examples of the combination of invoke task with build tags and `--bundle` or `--no-bundle` would help clarify the usage, as well as a couple of examples of running the agent using the `DD_BUNDLED_AGENT` env variable would help for the transition 😄

Reply:
While it's true that it changes how we internally build the agent and how we run it, it should be transparent : the command doesn't change and the executables are at the same location and one should not have to deal with the DD_BUNDLED_AGENT variable (by the way, at first I wanted to apply to single binary to packaging only but it was observed that using the same method for dev and released artifacts would reduce the risk of unnoticed bug, which totatally makes sense).

But adding a couple of examples surely doesn't hurt, so I'll update the PR right away :-)

Reply:
> by the way, at first I wanted to apply to single binary to packaging only but it was observed that using the same method for dev and released artifacts would reduce the risk of unnoticed bug, which totatally makes sense

Totally agree 😄 

> But adding a couple of examples surely doesn't hurt, so I'll update the PR right away :-)

Thanks you so much for doing it